docs/source/project_documentation/dev_guide/aac_constraints.md

```diff
@@ -0,0 +1,395 @@
+
+Generated implementation stub:
+
+```python
+"""The AaC Exclusive fields plugin implementation module."""
+# NOTE: It is safe to edit this file.
+# This file is only initially generated by aac gen-plugin, and it won't be overwritten if the file already exists.
+
+# There may be some unused imports depending on the definition of the plugin...but that's ok
+from aac.execute.aac_execution_result import (
+    ExecutionResult,
+    ExecutionStatus,
+    ExecutionMessage,
+    MessageLevel,
+)
+
+from aac.context.language_context import LanguageContext
+from aac.context.definition import Definition
+from aac.context.source_location import SourceLocation
+from typing import Any
+
+
+plugin_name = "Exclusive fields"
+
+
+def mutually_exclusive_fields(
+    instance: Any, definition: Definition, defining_schema, fields
+) -> ExecutionResult:
+    """Business logic for the Mutually exclusive fields constraint."""
+
+    # TODO: rewrite the code below to implement your constraint logic
+    # you can get the source and location from the definition for messages
+    status = ExecutionStatus.GENERAL_FAILURE
+    messages: list[ExecutionMessage] = []
+    error_msg = ExecutionMessage(
+        "The Mutually exclusive fields constraint for the Exclusive fields plugin has not been implemented yet.",
+        MessageLevel.ERROR,
+        definition.source,
+        None,
+    )
+    messages.append(error_msg)
+
+    return ExecutionResult(plugin_name, "Mutually exclusive fields", status, messages)
+
+```
+
+Generated test stub:
+```python
+from unittest import TestCase
+from typing import Tuple
+from click.testing import CliRunner
+from aac.execute.command_line import cli, initialize_cli
+from aac.execute.aac_execution_result import ExecutionStatus
+
+
+from aac.plugins.exclusive_fields.exclusive_fields_impl import plugin_name
+
+
+from aac.plugins.exclusive_fields.exclusive_fields_impl import mutually_exclusive_fields
+
+
+class TestExclusivefields(TestCase):
+    def test_mutually_exclusive_fields(self):
+        # TODO: Write tests for mutually_exclusive_fields
+        self.fail("Test not yet implemented.")
+```
+
+Since this is not a universal constraint, you must declare it where appropriate.  There's not a current example of the "Mutually Exclusive" constraint in the current AaC baseline, so let's switch over to the "If true then empty" constraint to see how schema constraints are allocated.  In this example, the constraint is used to check the boolean value of the `universal` field and if true, ensure there are no values provided for the `arguments` field.  Everything described above for plugin-generation applies here as well.  What we really want to focus on is the assignment of the schema constraint, including the arguments provided for the constraint.
```

Comment:
Universal constraints can't have arguments because they are not "assigned" within a specific model context.  Without an "assignment" there's nowhere to provide contextual parameters.  Logically, these are addressing model invariants rather than spceific content within a portion of a model.  If a constraint needs a contextual argument it's probably a schema constraint and would be "assigned" where appropriate.

Reply:
Language tweaked to try and be clearer.